FirebaseAppCheck/Sources/AppAttestProvider/API/FIRAppAttestAPIService.m

```diff
@@ -56,8 +59,83 @@ - (instancetype)initWithAPIService:(id<FIRAppCheckAPIServiceProtocol>)APIService
 }
 
 - (nonnull FBLPromise<NSData *> *)getRandomChallenge {
-  // TODO: Implement.
-  return [FBLPromise resolvedWith:nil];
+  NSString *URLString =
+      [NSString stringWithFormat:@"%@/projects/%@/apps/%@:generateAppAttestChallenge",
+                                 self.APIService.baseURL, self.projectID, self.appID];
+  NSURL *URL = [NSURL URLWithString:URLString];
+
+  return [FBLPromise onQueue:[self defaultQueue]
+                          do:^id _Nullable {
+                            return [self.APIService sendRequestWithURL:URL
+                                                            HTTPMethod:@"POST"
+                                                                  body:nil
+                                                     additionalHeaders:nil];
+                          }]
+      .then(^id _Nullable(GULURLSessionDataResponse *_Nullable response) {
+        return [self randomChallengeWithAPIResponse:response];
+      });
+}
+
+- (FBLPromise<NSData *> *)randomChallengeWithAPIResponse:(GULURLSessionDataResponse *)response {
+  return [FBLPromise onQueue:[self defaultQueue] do:^id _Nullable{
+    NSError *error;
+
+    NSData *randomChallenge = [self randomChallengeFromResponseBody:response.HTTPBody error:&error];
+
+    return randomChallenge ?: error;
+  }];
+}
+
+- (NSData *)randomChallengeFromResponseBody:(NSData *)response
+                                      error:(NSError **)outError {
```

Comment:
I was trying to follow what was done for App Check token (see [here](https://github.com/firebase/firebase-ios-sdk/blob/f2b77f0dce2def4037806b705c7608cfb4e24298/FirebaseAppCheck/Sources/Core/APIService/FIRAppCheckAPIService.m#L156)). Im unsure if it is a better practice to wrap the random challenge in a custom type (e.g. AppAttestRandomChallenge) so that the parsing done in `randomChallengeFromResponseBody:error:` can be done in the initializer like it is done in [AppCheckToken+Response category](https://github.com/firebase/firebase-ios-sdk/blob/appcheck-appattest-main/FirebaseAppCheck/Sources/Core/APIService/FIRAppCheckToken%2BAPIResponse.m)

Reply:
As we are interested in the challenge data only, not having a specific type is fine. Placing parsing of the response here looks also fine especially since we will simplify it a bit.